server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -160,17 +165,26 @@ public String executor() {
                 return ThreadPool.Names.GENERIC;
             }
         };
+
         try {
             // Will be replaced with ExtensionTokenProcessor and PrincipalIdentifierToken classes from feature/identity
             final String extensionTokenProcessor = "placeholder_token_processor";
             final String requestIssuerIdentity = "placeholder_request_issuer_identity";
 
+            this.allowList = List.of("Content-Type");
+            this.denyList = List.of("Authorization", "Proxy-Authorization");
+
+            Map<String, List<String>> filteredHeaders = headers.entrySet()
+                .stream()
+                .filter(e -> !denyList.contains(e.getKey()))
+                .filter(e -> allowList.contains("*") || allowlist.contains(e.getKey()))
+                .collect(Collectors.toMap());
+
             transportService.sendRequest(
                 discoveryExtensionNode,
                 ExtensionsManager.REQUEST_REST_EXECUTE_ON_EXTENSION_ACTION,
-                // HERE BE DRAGONS - DO NOT INCLUDE HEADERS
-                // SEE https://github.com/opensearch-project/OpenSearch/issues/4429
```

Comment:
What are your thoughts about this issue / why remove it from the code?

Note; I think it could be appropriate to move these comment where the header filter is occurring that you have added.